packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -279,6 +298,98 @@ def draw_image(self, gc, x, y, im, transform=None):
         self.ctx.drawImage(in_memory_canvas, x, y, w, h)
         self.ctx.restore()
 
+    def _get_font(self, prop):
+        key = hash(prop)
+        font_value = self.fontd.get(key)
+        if font_value is None:
+            fname = findfont(prop)
+            font_value = self.fontd.get(fname)
+            if font_value is None:
+                font = FT2Font(str(fname))
+                font_file_name = fname[fname.rfind('/')+1:]
+                font_value = font, font_file_name
+                self.fontd[fname] = font_value
+            self.fontd[key] = font_value
+        font, font_file_name = font_value
+        font.clear()
+        font.set_size(prop.get_size_in_points(), self.dpi)
+        return font, font_file_name
+
+    def get_text_width_height_descent(self, s, prop, ismath):
+        if ismath:
+            image, d = self.mathtext_parser.parse(s, self.dpi, prop)
+            w, h = image.get_width(), image.get_height()
+        else:
+            font, _ = self._get_font(prop)
+            font.set_text(s, 0.0, flags=LOAD_NO_HINTING)
+            w, h = font.get_width_height()
+            w /= 64.0
+            h /= 64.0
+            d = font.get_descent() / 64.0
+        return w, h, d
+
+    def _draw_math_text(self, gc, x, y, s, prop, angle):
+        rgba, descent = self.mathtext_parser.to_rgba(s, gc.get_rgb(), self.dpi,
+                                                     prop.get_size_in_points())
+        height, width, _ = rgba.shape
+        angle = math.radians(angle)
+        if angle != 0:
+            self.ctx.save()
+            self.ctx.translate(x, y)
+            self.ctx.rotate(-angle)
+            self.ctx.translate(-x, -y)
+        self.draw_image(gc, x, -y-descent, np.flipud(rgba))
+        if angle != 0:
+            self.ctx.restore()
+
+    def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
+        def _load_font_into_web(loaded_face):
+            document.fonts.add(loaded_face)
+            window.font_loaded = True
+            _font_set.add(font_face_arguments)
+            self.fig.draw_idle()
+
+        if ismath:
+            self._draw_math_text(gc, x, y, s, prop, angle)
+            return
+        angle = math.radians(angle)
+        width, height, descent = \
+            self.get_text_width_height_descent(s, prop, ismath)
+        x -= math.sin(angle) * descent
+        y -= math.cos(angle) * descent - self.ctx.height
+        font_size = \
+            self.points_to_pixels(prop.get_size_in_points())
+
+        _, font_file_name = self._get_font(prop)
+
+        font_face_arguments = (prop.get_name(), 'url({0})'.format(
```

Comment:
Let's add a comment here that this is loading the font into the browser's environment and why we have to avoid loading the same font multiple times.

Reply:
Done!